talisker/sentry.py

```diff
@@ -137,7 +126,12 @@ def log_client(client, from_env=False):
         return
 
     # base_url shouldn't have secrets in, but just in case, clean it
-    public_dsn = client.remote.get_public_dsn()
+    try:
+        public_dsn = client.remote.get_public_dsn()
+    except Exception:
+        # people patch is_enabled in tests, don't fail here if there is no DSN
```

Comment:
This is usually too broad, it'll catch AttributeError, TypeError etc. Is it possible to catch only the case when there's no DSN set? Perhaps by looking at the configuration?